src/templates/React InstantSearch widget/vite.config.ts

```diff
@@ -0,0 +1,33 @@
+import path from 'path';
+
+import reactRefresh from '@vitejs/plugin-react-refresh';
+import vite from 'vite';
+
+// import { defineConfig } from 'vite' -> fails when type:module in package.json
+// it seems related to https://github.com/vitejs/vite/issues/1560
+// so far this is the workaround.
+const { defineConfig } = vite;
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [reactRefresh()],
+  build: {
+    lib: {
+      entry: path.resolve(__dirname, 'src/index.tsx'),
+      name: 'SampleCustom',
+      fileName: 'index',
+      formats: ['es', 'cjs', 'umd'],
+    },
+    rollupOptions: {
+      // make sure to externalize deps that shouldn't be bundled
+      // into your library
+      external: ['react', 'react-dom', 'react-instantsearch-dom'],
+      output: {
+        globals: {
```

Comment:
We don't need to specify ReactDOM here in case the widget relies on it? (e.g., if it uses React Portals)

Reply:
You're right. I forgot it. [`9856a5b` (#456)](https://github.com/algolia/create-instantsearch-app/pull/456/commits/9856a5b7938e2ffc5787f86715714ff42e881a55)